cmd/soroban-cli/src/config/mod.rs

```diff
@@ -47,7 +47,12 @@ pub struct Args {
     #[command(flatten)]
     pub network: network::Args,
 
-    #[arg(long, visible_alias = "source", env = "STELLAR_ACCOUNT")]
+    #[arg(
+        long,
+        visible_alias = "source",
+        env = "STELLAR_ACCOUNT",
+        default_value = ""
```

Comment:
I think we should make the type an `Option<String>` instead of having an empty string value. The empty string value is not common to see in Rust used in this way, and the interface is not so nice because the `--help` prints out 
    
    Default value: ``

Reply:
+1 It was my suggestion because we were using `FromStr`. I think perhaps the Address enum could add a `Default`, case.

```rust
#[derive(Clone, Debug, Default)]
pub enum Address {
    MuxedAccount(xdr::MuxedAccount),
    AliasOrSecret(String),
    #[default]
    Config,
}
```

Then we can do `self.source.unwrap_or_default()`.

Reply:
Using `Option<String>` means we'll need to drop the current `Address` type, right? This is a "raw" value that's passed to the value parser.

I'll explore Willem's suggestion with the `FromStr` trait.

Reply:
Btw, without setting a default value on the `arg` macro, the `--source-account` will always be required. We can explore using `default_value_t`, but we'd need to return a `Address`-typed object. I don't think we can get away with not setting a default value with clap.

Reply:
About the default value, clap has `hide_default_value = true`, which hides it completely from the `--help` output. Unfortunately, clap-markdown doesn't respect it ([already have a ticket upstream for it](https://github.com/ConnorGray/clap-markdown/issues/33)).

Reply:
Sorry, my reference to String above was a typo. I meant `Option<Address>`.

Reply:
> ```rust
> #[derive(Clone, Debug, Default)]
> pub enum Address {
>     MuxedAccount(xdr::MuxedAccount),
>     AliasOrSecret(String),
>     #[default]
>     Config,
> }
> ```

I would avoid ☝🏻 because it makes the type itself recursive. Anywhere you receive the `Address` the consuming code would need to assume it could be `Address::Config` and need to recurse to get a "resolved" `Address` that wasn't `Address::Config`, but then the type still contains that type so any consuming code would be forced to handle it.

My main issue with the approach in the PR at the moment is really not to do with the type at all, but to do with the unpacking/resolving logic happening inside the type. I think the unpacking belongs outside the type, mostly because I'd prefer types to be largely data, rather than integrated with the presentation layer. But I understand it's convenient to put it there, and sort of awkward to do it the Option<Address> way anwyay, and we already do resolution inside from_str anyway, so LGTM. 👍🏻

Reply:
Going to merge it as is. If we can improve it, the improvement should be a no-op non-functional refactor and we can follow up with it.

Reply:
The only place where this `Address` is pulled from the config is for the source account, so we could just handle the option is the `source_account()`, method. If `self.source_account` is None, then we use the config, otherwise it will be the `Address` type and it resolves as normal. I do agree that using the empty string as a special case is not very idiomatic rust and Option is the preferred way to do it.  Plus my recent work adding a `ContractAddress` type make me realize that we need a unified approach between it and `Address` because in contract invoke an `ScAddress` could be either! So more reason not to handle the config at all in the `Address` type.